hm_pyhelper/miner_param.py

```diff
@@ -179,34 +184,37 @@ def get_mac_address(path):
     return file.readline().strip().upper()
 
 
-REGION_OVERRIDE_KEY = 'REGION_OVERRIDE'
-REGION_FILEPATH = '/var/pktfwd/region'
-REGION_INVALID_SLEEP_SECONDS = 30
-REGION_FILE_MISSING_SLEEP_SECONDS = 60
-
-
-class MalformedRegionException(Exception):
-    pass
-
-
-@retry(MalformedRegionException, delay=REGION_INVALID_SLEEP_SECONDS, logger=logger) # noqa
-@retry(FileNotFoundError, delay=REGION_FILE_MISSING_SLEEP_SECONDS, logger=logger) # noqa
-def get_region():
+@retry(MalformedRegionException, delay=REGION_INVALID_SLEEP_SECONDS, logger=LOGGER) # noqa
+@retry(FileNotFoundError, delay=REGION_FILE_MISSING_SLEEP_SECONDS, logger=LOGGER) # noqa
+def retry_get_region(region_override, region_filepath):
     """
-    Return the region from the environment or parse file created by hm-miner.
+    Return the override if it exists, or parse file created by hm-miner.
+    region_override is the actual value,
+    not the name of the environment variable.
     Retry if region in file is malformed or not found.
     """
-    region = os.getenv(REGION_OVERRIDE_KEY, False)
-    if region:
-        return region
+    if region_override:
+        return region_override
 
-    logger.debug("No REGION_OVERRIDE defined, will retrieve from miner.")
-    with open(REGION_FILEPATH) as region_file:
+    LOGGER.debug("No region override set (value = %s), will retrieve from miner." % region_override)  # noqa: E501
```

Comment:
Is the idea here that we can pass other values to the retry_get_region command for different services should we need to?

Rather than hard coding it with the REGION_OVERRIDE and REGION_FILEPATH env variables?

Reply:
Portability to other services is one reason. The bigger reason is I think its hard to maintain services with environment variables that are read in the depths of the code. I prefer to parameterize everything and make it up to the caller to decide where those parameters come from. Both in hm-config and hm-pktfwd refactors, `__main__.py` acts as a single entry point where the environment is read (eg: https://github.com/NebraLtd/hm-pktfwd/pull/59/files#diff-a6ac805adb62e68e9273e1c2fc120902571d800d5596a1df4dcb316b303d520aR30).

So if we decided to read all these values from a `.env` file in the future instead of the actual environment, we can just change the main function without having to dig through the code to find all the places the environment is read.